### PR TITLE
[ServiceBus] `az servicebus namespace`: Add new commands private-endpoint-connection and private-link-resource commands to manage private endpoint connection and private link resources

### PR DESCRIPTION
**Pull request by v-Ajnava**
**Description**

>Added zone-redundant parameter to namespace create command
>Added disable-local-auth parameter to namespace create and update command
>Added new private-endpoint-connection and private-link-resource commands for namespace

**Testing Guide**

### Approve 

> az servicebus namespace private-endpoint-connection approve
example: 
az servicebus namespace private-endpoint-connection approve -g myRg --namespace-name mynamespace --name myconnection

### Delete 

> az servicebus namespace private-endpoint-connection delete
example: 
az servicebus namespace private-endpoint-connection delete -g myRg --namespace-name mynamespace --name myconnection

### Delete 

> az servicebus namespace private-endpoint-connection reject
example: 
az servicebus namespace private-endpoint-connection reject -g myRg --namespace-name mynamespace --name myconnection

### show

> az servicebus namespace private-endpoint-connection show
example: 
az servicebus namespace private-endpoint-connection show -g myRg --namespace-name mynamespace --name myconnection

### show

> az servicebus namespace private-link-resource show
example: 
az servicebus namespace private-link-resource show --namespace-name mynamespace -g MyResourceGroup

**History Notes**

[ServiceBus] `az servicebus namespace create/update`: Add `--disable-local-auth` to enable or disable SAS authentication
[ServiceBus] `az servicebus namespace private-endpoint-connection/private-link-resource`: New command groups

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ServiceBus

**Review by wangzelin007 — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/servicebus/custom.py [hunk 1]
```diff
@@ -618,3 +637,50 @@ def cli_returnnsdetails(cmd, resource_group_name, namespace_name, max_size_in_me
...
+    PrivateEndpointServiceConnectionStatus = cmd.get_models('PrivateLinkConnectionStatus')
+
+    private_endpoint_connection = client.get(resource_group_name=resource_group_name, namespace_name=namespace_name,
+                                             private_endpoint_connection_name=private_endpoint_connection_name)
+
+    old_status = private_endpoint_connection.private_link_service_connection_state.status
+    if old_status != "Approved" or not is_approved:
+        private_endpoint_connection.private_link_service_connection_state.status = PrivateEndpointServiceConnectionStatus.APPROVED\
+            if is_approved else PrivateEndpointServiceConnectionStatus.REJECTED
+        private_endpoint_connection.private_link_service_connection_state.description = description
```
Comment (damodaravadhani) [inline src/azure-cli/azure/cli/command_modules/servicebus/custom.py:657]: We should be updating description regardless of what updated status is right ? Shouldn't this be outside if block. and we should be updating it only if the user has given description and not other times.

**Review by damodaravadhani — CHANGES_REQUESTED** [review]

**Comment by v-Ajnava** [comment]
Hi @wangzelin007 and @evelyn-ys  can you please review the PR and help in in understanding the failure of test in latest profile.

**Comment by damodaravadhani** [comment]
@yonzhan , can you add the May release build label to this PR ?

**Review by evelyn-ys — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Comment by damodaravadhani** [comment]
image
@evelyn-ys , the pipelines point to a certain azure/cli/command_modules/network/tests/latest/test_network_commands.py::NetworkVirtualApplianceScenarioTest::test_network_virtual_appliance and asks us to push the recording although it doesn't use service bus in the test. Do you happen to know why this is the case? Thanks.

**Comment by evelyn-ys** [comment]
@necusjz Can you help with the network test failure?

**Review by zhoxing-ms — APPROVED** [review]

**Comment by damodaravadhani** [comment]
Hey @evelyn-ys, could we please merge this PR ? Thanks.

**Pull request merged**